source/scpp/src/scp/QuorumSetUtils.cpp

```diff
@@ -33,24 +34,50 @@ class QuorumSetSanityChecker
     bool mIsSane;
     size_t mCount{0};
 
-    bool checkSanity(SCPQuorumSet const& qSet, int depth);
+    bool checkSanity(SCPQuorumSet const& qSet, int depth, const char** reason);
 };
 
 QuorumSetSanityChecker::QuorumSetSanityChecker(SCPQuorumSet const& qSet,
-                                               bool extraChecks)
+                                               bool extraChecks,
+                                               const char** reason)
     : mExtraChecks{extraChecks}
 {
-    mIsSane = checkSanity(qSet, 0) && mCount >= 1 && mCount <= 1000;
+    mIsSane = checkSanity(qSet, 0, reason);
```

Comment:
You can simplify the code by making the `reason` optional on the API, but required in the functions (IOW provides a dummy variable if `reason` is null so you avoid bounds check).